tests/scalers/splunk_observability_test/splunk_observability_test.go

```diff
@@ -0,0 +1,185 @@
+//go:build e2e
+// +build e2e
+
+package splunk_observability_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+const (
+	testName = "splunk-observability-test"
+)
+
+var (
+	testNamespace          = fmt.Sprintf("%s-ns", testName)
+	deploymentName         = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName       = fmt.Sprintf("%s-so", testName)
+	authName               = fmt.Sprintf("%s-auth", testName)
+	duration               = "10"
+	maxReplicaCount        = 10
+	minReplicaCount        = 1
+	scaleInTargetValue     = "400"
+	scaleInActivationValue = "1.1"
+)
+
+type templateData struct {
+	TestNamespace         string
+	DeploymentName        string
+	ScaledObjectName      string
+	AuthName              string
+	Duration              string
+	MinReplicaCount       string
+	MaxReplicaCount       string
+	TargetValue           string
+	ActivationTargetValue string
+}
+
+const (
+	authTemplate = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: splunk-secrets
+  namespace: {{.TestNamespace}}
+data:
+  accessToken: YW1JeUpqVHRJd185cDhOWG01X21KQQ==  # one time through-away access token used just for testing
+  realm: dXMw
+---
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: keda-trigger-auth-splunk-secret
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  - parameter: accessToken
+    name: splunk-secrets
+    key: accessToken
+  - parameter: realm
+    name: splunk-secrets
+    key: realm
+`
+
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+  namespace: {{.TestNamespace}}
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  replicas: 1 
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: keda
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: nginx
+  pollingInterval: 3
+  cooldownPeriod: 1
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  triggers:
+  - type: splunk-observability
+    metricType: Value
+    metadata:
+      query: "data('fdse-1989-tenable-test-metric').publish()"
```

Comment:
This should probably be a more generic metric name and not pertaining to any jiras